Packs/GCP-Enrichment-Remediation/Scripts/GCPOffendingFirewallRule/GCPOffendingFirewallRule.py

```diff
@@ -0,0 +1,128 @@
+import demistomock as demisto  # noqa: F401
+from CommonServerPython import *  # noqa: F401
+from typing import Dict, Any
+import traceback
+
+
+''' STANDALONE FUNCTION '''
+
+
+def test_range(port_range: str, port: str) -> bool:
+    """
+    Breaks a string port range (i.e. '20-25') into integers for comapirson
+    Args:
+        port_range (str): string based port range from the GCP firewall rule.
+        port (str): port supplied in script args.
+
+    Returns:
+        bool: whether there was a match between the supplied port in args and the supplied range.
+    """
+    split_range = port_range.split('-')
+    if int(port) >= int(split_range[0]) and int(port) <= int(split_range[1]):
+        return True
+    return False
+
+
+def test_match(port: str, protocol: str, rule: Dict, network_tags: list) -> bool:
+    """
+    Breaks a string port range (i.e. '20-25') into integers for comapirson
+    Args:
+        port (str): port supplied in script args.
+        protocol (str): protocol supplied in script args.
+        rule (Dict): GCP firewall rule pulled from integration command.
+        network_tags (list): list of network tags (can be empty).
+
+    Returns:
+        bool: whether there was a match between the supplied port, protocol
+        and possible target tag combination and the GCP firewall rule.
+    """
+    # Match rule needs to be direction ingress, source from internet (0.0.0.0/0), enabled and an allow rule.
+    if rule.get('direction') == 'INGRESS' and '0.0.0.0/0' in rule.get('sourceRanges', []) \
+       and rule.get('disabled') is False and 'allowed' in rule.keys():
+        # Test if targetTags are relevant or not (if show up in keys or tag match)
+        target_tags_verdict = ('targetTags' not in rule.keys() or len(set(rule.get('targetTags', [])) & set(network_tags)) > 0)
+        for entry in rule.get('allowed', []):
+            # Match is all protocol AND either no target tags OR target tags match
+            if entry.get('IPProtocol') == 'all' and target_tags_verdict:
+                return True
+            # Complicated because just {'IPProtocol': 'udp'} means all udp ports
+            # therefore if protocol match but no ports, this is a match
+            elif entry.get('IPProtocol') == protocol.lower() and 'ports' not in entry.keys():
```

Comment:
```suggestion
            elif entry.get('IPProtocol') == protocol.lower() and 'ports' not in entry:
```
https://docs.astral.sh/ruff/rules/in-dict-keys/